doc/en/pools.rst

```diff
@@ -374,3 +374,62 @@ The code above specifies a collections of parameters in `@task_target`, and each
 entry will be used create one task - thus 3 tasks will be created from the target.
 
 For a complete and downloadable example, see :ref:`here <example_discover>`.
+
+
+.. _auto_part:
+
+Auto-Part and Smart Scheduling
+------------------------------
+
+Testplan can automatically determinate the optimal number of parts for multitests from historical runtime data.
```

Comment:
This feature allows schedule_all() to optimize testplan overall execution time based on historical runtime data. It is enabled by providing runtime data like following via ``--runtime-data`` command line argument:

...

The optimization goal is to create just enough number of pool workers and allow all tests to finish as soon as possible. This is achieved by 3 technics:

1) auto-part: automatically slice multitests discovered from @task_target with ``multitest_parts="auto"`` argument into optimal number of parts subject to ``auto_part_runtime_limit`` - default to 30 minutes

...


2) weight-based scheduling: each multitest (part) will be associated with a weight value that represents historical runtime
3) auto-size: if the target pool is specified to have "auto" size, schedule_all() will calculate a right number of pool workers so that all tests finishes within ``plan_runtime_target`` - default to 30 minutes. 



that are decorated with @task_target(multitest_parts="auto")

This feature aims to optimize testplan overall execution time optimization leveraging the historical runtime data. 

If historical runtime data is provided by user, schedule_all() can automatically minimize testplan execution time by automatically determine optimal number of parts for multitest, create pool of the right size, and schedule tasks in the right order -- based task creation and scheduling based historical runtime time. on schedule_all() is able to 
When schedule_all() is used, user can in
This feature aims to optimize testplan execution time when schedule_all() is used by leveraging historical runtime data.